custom_components/unifiprotect/binary_sensor.py

```diff
@@ -268,6 +325,20 @@ def _async_get_extra_attrs(self) -> dict[str, Any]:
     def _async_update_device_from_protect(self) -> None:
         super()._async_update_device_from_protect()
 
+        if self.entity_description.key.startswith(_KEY_DISK_HEALTH):
+            assert isinstance(self.device, NVR)
+            assert self._index is not None
+
+            disks = self.device.system_info.storage.devices
+            self._attr_available = self._attr_available and len(disks) > self._index
+            if self._attr_available:
+                disk = disks[self._index]
+                self._attr_is_on = not disk.healthy
+
+                self._extra_state_attributes = {ATTR_MODEL: disk.model}
```

Comment:
Don't not need to check `_attr_available`, the state will be calculated automatically
```suggestion
             disk = disks[self._index]
             self._attr_is_on = not disk.healthy
             self._extra_state_attributes = {ATTR_MODEL: disk.model}
```

Reply:
This is on purpose to avoid an index error. I am setting `_attr_avaiable` right above with if the index exists.

Reply:
https://github.com/briis/unifiprotect/pull/379/files#r762597650